src/anyon/run_jobs.jl

```diff
@@ -0,0 +1,490 @@
+    generation    ::String  ="Yukon"
+    serial_number ::String  ="ANYK202201"
+    printout_delay::Real    =200. # milliseconds between get_status() printouts
+end
+
+get_client(qpu_service::AnyonQPU)=qpu_service.client
+get_printout_delay(qpu_service::AnyonQPU)=qpu_service.printout_delay
+
+function Base.show(io::IO, qpu::AnyonQPU)
+    println(io, "Quantum Processing Unit:")
+    println(io, "   manufacturer:  $(qpu.manufacturer)")
+    println(io, "   generation:    $(qpu.generation) ")
+    println(io, "   serial_number: $(qpu.serial_number) ")
+end
+
+"""
+    VirtualQPU
+
+A data structure to represent a Quantum Simulator.  
+# Fields
+- `developers   ::String`   -- Simulator developers.
+- `package      ::String`   -- name of the underlying library package.
+
+# Example
+```jldoctest
+julia> qpu=VirtualQPU()
+Quantum Simulator:
+   developers:  Anyon Systems Inc.
+   package:     Snowflake.jl
+
+
+```
+"""
+Base.@kwdef struct VirtualQPU <: AbstractQPU
+    developers  ::String  ="Anyon Systems Inc."
+    package     ::String  ="Snowflake.jl"
+end
+
+function Base.show(io::IO, qpu::VirtualQPU)
+    println(io, "Quantum Simulator:")
+    println(io, "   developers:  $(qpu.developers)")
+    println(io, "   package:     $(qpu.package)")
+end
+
+
+function format_response(response::HTTP.Messages.Response)
+    formatted_response=Dict(
+        "version"   =>response.version, 
+        "status"    =>response.status, 
+        "headers"   =>response.headers, 
+    )
+
+    # convert response body from binary to ASCII
+    read_buffer=IOBuffer(reinterpret(UInt8, response.body))
+    formatted_response["body"]=String(readuntil(read_buffer, 0x00))
+
+    return formatted_response
+end
+
+"""
+    run_job(qpu::AnyonQPU, circuit::QuantumCircuit,num_repetitions::Integer)
+
+Run a circuit computation on a `QPU` service, repeatedly for the specified 
+number of repetitions (num_repetitions). Returns the histogram of the 
+completed circuit calculations, or an error message.
+
+# Example
+```jldoctest
+julia> c = Client(host="http://example.anyonsys.com",user="test_user",access_token="not_a_real_access_token",requestor=MockRequestor());
+  
+julia> qpu=AnyonQPU(client=c);
+
+julia> run_job(qpu,QuantumCircuit(qubit_count=3,gates=[sigma_x(3),control_z(2,1)]) ,100)
+Circuit submitted: circuitID returned: 8050e1ed-5e4c-4089-ab53-cccda1658cd0
+
+status: succeeded
+Dict{String, Int64} with 1 entry:
+  "001" => 100
+
+```
+"""
+function run_job(qpu::AnyonQPU, circuit::QuantumCircuit,num_repetitions::Integer)::Dict{String,Int}
+    
+    client=get_client(qpu)
+
+    circuitID=submit_circuit(client,circuit,num_repetitions)
+
+    println("Circuit submitted: circuitID returned: $circuitID\n")
+
+    status=get_status(client,circuitID;)
+
+    while true
+        
+        println("status: $(status["type"])")
+        
+        if !(status["type"] in ["queued","running"])
+            break
+        end
+
+        sleep(get_printout_delay(qpu)/1000) 
```

Comment:
Let's not make the wait between statuses dependent on print duration. We could want the latest status to be printed only every second but query the status every 100ms.